src/init/pkg_installer_mac.sh

```diff
@@ -0,0 +1,132 @@
+
+# Generate wazuh home directory tree to backup
+FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/active-response)
+FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/bin)
+FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/etc)
+FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/lib)
+FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/queue)
+[ -d "./ruleset" ] && FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/ruleset)
+[ -d "./wodles" ] && FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/wodles)
+[ -d "./agentless" ] && FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/agentless)
+[ -d "./logs/ossec" ] && FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/logs/ossec)
+[ -d "./var/selinux" ] && FOLDERS_TO_BACKUP+=(${CURRENT_DIR}/var/selinux)
+
+for dir in "${FOLDERS_TO_BACKUP[@]}"; do
+    mkdir -p "${TMP_DIR_BACKUP}${dir}"
+    cp -a ${dir}/* "${TMP_DIR_BACKUP}${dir}"
+done
+
+if [ -f /etc/ossec-init.conf ]; then
+    mkdir -p ./tmp_bkp/etc
+    cp -p /etc/ossec-init.conf ./tmp_bkp/etc
+fi
+
+# Save service file
+mkdir -p "${TMP_DIR_BACKUP}/Library/LaunchDaemons"
+cp -a /Library/LaunchDaemons/com.wazuh.agent.plist "${TMP_DIR_BACKUP}/Library/LaunchDaemons"
+
+mkdir -p "${TMP_DIR_BACKUP}/Library/StartupItems/WAZUH"
+cp -a /Library/StartupItems/WAZUH/* ${TMP_DIR_BACKUP}/Library/StartupItems/WAZUH
+
+# Saves modes and owners of the directories
+BACKUP_LIST_FILES=$(find "${TMP_DIR_BACKUP}/" -type d)
+
+while read -r line; do
+    org=$(echo "${line}" | awk "sub(\"${TMP_DIR_BACKUP}\",\"\")")
+    owner=$(stat -f %Su ${org})
+    group=$(stat -f %Sg ${org})
+    mode=$(stat -f %A ${org})
+    chown $owner:$group $line
+    chmod $mode $line
+done <<< "$BACKUP_LIST_FILES"
+
+# Generate Backup
+mkdir -p ./backup
+tar czf ./backup/backup_[${BDATE}].tar.gz -C ./tmp_bkp . >> ./logs/upgrade.log 2>&1
+rm -rf ./tmp_bkp/
+
+# Installing upgrade
+echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade started." >> ./logs/upgrade.log
+installer -pkg ./var/upgrade/wazuh-agent* -target / >> ./logs/upgrade.log 2>&1
+
+# Check installation result
+RESULT=$?
+
+echo "$(date +"%Y/%m/%d %H:%M:%S") - Installation result = ${RESULT}" >> ./logs/upgrade.log
+
+# Wait connection
+status="pending"
+COUNTER=30
+while [ "$status" != "connected" -a $COUNTER -gt 0 ]; do
+    . ./var/run/wazuh-agentd.state >> ./logs/upgrade.log 2>&1
+    sleep 1
+    COUNTER=$[COUNTER - 1]
+    echo "$(date +"%Y/%m/%d %H:%M:%S") - Waiting connection... Status = "${status}". Remaining attempts: ${COUNTER}." >> ./logs/upgrade.log
+done
+
+# Check connection
+if [ "$status" = "connected" -a $RESULT -eq 0 ]; then
+    echo "$(date +"%Y/%m/%d %H:%M:%S") - Connected to manager." >> ./logs/upgrade.log
+    echo -ne "0" > ./var/upgrade/upgrade_result
+    echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade finished successfully." >> ./logs/upgrade.log
+else
+    # Restore backup
+    echo "$(date +"%Y/%m/%d %H:%M:%S") - Upgrade failed. Restoring..." >> ./logs/upgrade.log
+
+    # Cleanup before restore
+    CONTROL="./bin/wazuh-control"
+    if [ ! -f $CONTROL ]; then
+        CONTROL="./bin/ossec-control"
+    fi
```

Comment:
From v4.2 this is always `wazuh-control`

Reply:
Fixed.